src/components/elements/upload/Uploader.tsx

```diff
@@ -0,0 +1,385 @@
+import UploadFileIcon from '@mui/icons-material/UploadFile';
+import {
+  alpha,
+  Box,
+  Card,
+  Divider,
+  LinearProgress,
+  Link,
+  Stack,
+  Typography,
+} from '@mui/material';
+import { visuallyHidden } from '@mui/utils';
+import { compact, flatten, isEmpty, sortBy, uniq } from 'lodash-es';
+import React, { useCallback, useMemo, useRef, useState } from 'react';
+import { Accept, useDropzone } from 'react-dropzone';
+import useDirectUpload from './useDirectUpload';
+
+import FilePreviewIcon from '@/components/elements/upload/FilePreviewIcon';
+import FileRecordSummary from '@/components/elements/upload/fileSummary/FileRecordSummary';
+import FileUploadSummary from '@/components/elements/upload/fileSummary/FileUploadSummary';
+import { DirectUpload, FileFieldsFragment } from '@/types/gqlTypes';
+import { ensureArray } from '@/utils/arrays';
+
+const DEFAULT_MAX_BYTES = 3000000;
+const IMAGE_FILE_TYPES = ['.png', '.jpg', '.jpeg', '.gif'];
+
+const getFileTypesFromAccept = (accept: Accept) => {
```

Comment:
File structure note: these helper methods could go in `modules/uploader/util.rb`